framework/src/modules/chain/blocks/blocks.js

```diff
@@ -505,14 +540,53 @@ class Blocks extends EventEmitter {
 		}
 	}
 
+	// PRE: Block has been validated and verified before
+	async _processReceivedBlockV2(block) {
+		try {
+			const newBlock = await this.blocksProcess.processBlock(
+				block,
+				this._lastBlock,
+				validBlock => this.broadcast(validBlock)
+			);
+
+			this.logger.debug(
+				`Successfully applied new received block id: ${block.id} height: ${
+					block.height
+				} round: ${this.slots.calcRound(
+					block.height
+				)} slot: ${this.slots.getSlotNumber(block.timestamp)} reward: ${
+					block.reward
+				} version: ${block.version}`
+			);
+			this._lastReceivedAndAppliedBlock.receivedTime = this._lastReceipt;
+			this._lastReceivedAndAppliedBlock.id = block.id;
+			this._lastBlock = newBlock;
+			this._isActive = false;
```

Comment:
already setting it to false in finally. We can remove it from here.

Reply:
Solved.